pkg/distributor/distributor.go

```diff
@@ -205,6 +212,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	f.DurationVar(&cfg.RemoteTimeout, "distributor.remote-timeout", 2*time.Second, "Timeout for downstream ingesters.")
 	f.BoolVar(&cfg.WriteRequestsBufferPoolingEnabled, "distributor.write-requests-buffer-pooling-enabled", false, "Enable pooling of buffers used for marshaling write requests.")
 	f.BoolVar(&cfg.LimitInflightRequestsUsingGrpcMethodLimiter, "distributor.limit-inflight-requests-using-grpc-method-limiter", false, "Use experimental method of limiting push requests.")
+	f.IntVar(&cfg.IngesterPushWorkerGoroutines, "distributor.ingester-push-worker-goroutines", 0, "Number of pre-allocated worker goroutines used to forward push requests to the ingesters. If 0, no workers will be used and a new goroutine will be spawned for each ingester push request. If not enough workers available, new goroutine will be spawned.")
```

Comment:
`-distributor.ingester-push-worker-goroutines` sounds like "_I am going to use at most this number of goroutines for ingester push requests_", whereas it means "_I will reuse this number of goroutines, and spawn new ones if it is needed_".
WDYT if we call this `-distributor.reusable-ingester-push-workers` or `-distributor.reusable-ingester-push-worker-goroutines`?

Reply:
Renamed to `distributor.reusable-ingester-push-workers` in https://github.com/grafana/mimir/pull/6660/commits/690cc0cc8de798e9834feea0cbff4ac95a9837a7

Thank you :pray:

Reply:
Thank you